src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -48,6 +58,59 @@ public HttpConnectionDispatcher(ConnectionManager manager, ILoggerFactory logger
             }
         }
 
+        private async Task<bool> AuthorizeAsync<TEndPoint>(HttpContext context) where TEndPoint : EndPoint
```

Comment:
Something like https://github.com/aspnet/Security/pull/1148 look reasonable?  It basically splits things into the authentication method, and authorization method (eventually this might have logic to distinguish between challenge/forbidden)